framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py

```diff
@@ -71,6 +77,190 @@ def run(self,inputDic):
     if len(inputDic)>1:
       self.raiseAnError(IOError, self.__class__.__name__ + ' Interfaced Post-Processor ' + str(self.name) + ' accepts only one dataObject')
 
+    #get actual data
+    inputDict = inputDic[0]['data']
+    #identify features
+    self.features = inputDict['outVars']# self.features = inputDict['output'][inputDict['output'].keys()[0]].keys()
+    #don't keep the pivot parameter in the feature space
+    if self.pivotParameter in self.features:
+      self.features.remove(self.pivotParameter)
+
+    #if output length (size of desired output history) not set, set it now
+    if self.outputLen is None:
+      self.outputLen = np.asarray(inputDict['output'][inputDict['output'].keys()[0]][self.pivotParameter])[-1]
+
+    ## Check if data is synchronized
+    referenceHistory = 0 # inputDict['output'].keys()[0]
+    referenceTimeAxis = inputDict[self.pivotParameter][referenceHistory]
```

Comment:
We have a method in the dataobject for checking axial alignment.  Can we replace this with that?

Reply:
this was performed before this restructuring. I think we should maximize timing later.